enhancements/network/ovs-flow-export-configuration.md

```diff
@@ -0,0 +1,267 @@
+emission-collection traffic to the local node.
+
+The collector can be deployed as a DaemonSet, to ensure that there will be an
+instance of the collector for each node.
+
+Exposing the collector endpoint as a Kubernetes ClusterIP Service would cause the traffic
+to be distributed across all the nodes (there is a `ServiceInternalTrafficPolicy`
+cluster-wide feature gate but, at the moment of writing this document, it is not implemented in
+OVN-Kubernetes, and it's beta in Kubernetes 1.22).
+
+To limit the flows' traffic to internal nodes, each collector should expose a node
+port, so each OVN instance will discover the Node IP (instead of the Service IP) and submit
+data to the NodeIP:NodePort address. The Node IP can be discovered.
+
+From the user-side configuration, the NOO has a custom resource definition that covers IPFix cache and sampling settings, deployment model (as a DaemonSet or as Deployment+Service) and a few other settings not relevant here.
+
+The NOO will generate a `ConfigMap` that is available for reading / watching to the CNO. The CNO would configure the OVN pods with the `ConfigMap` data.
+
+This solution ensures the NOO never interacts directly with OvS in order to avoid any configuration or state conflict with the CNO, and avoids creating a hard dependency in the CNO towards the NOO.
+
+Given the aforementioned details, following subsections depicts the work to do,
+grouped in different tasks.
+
+#### Task 1: setup a new ConfigMap that is readable by the CNO
+
+[Current `Network` operator configuration allows setting a static host:port address](
+https://docs.okd.io/latest/networking/ovn_kubernetes_network_provider/tracking-network-flows.html#nw-network-flows-object_tracking-network-flows).
+This information is forwarded to each OVN instance, on each cluster node. For example:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  exportNetworkFlows:
+    collectors:
+      - 192.168.1.99:2056
+```
+
+However, we need to set this address dynamically to ensure that each pod will forward
+the flows to the collector in its same node.
+
+While the above `Network` configuration is still valid for the current use cases, the NOO will generate a new `ConfigMap` that is exposed to the CNO, which
+will use it to configure the OVN pods. The CNO has to watch for any change in this configuration.
+
+![Interaction between CNO and NOO](./ovs-flow-export-configuration-CNO-NOO.png)
+
+The `ConfigMap` content adds up with any configuration that is already present in the Network CR. For example, if a user has configured an IP for netflow exports in the Network CR AND has configured a Netobs CR for the NOO, OvS will export to both destinations.
+
+Here is the new `ConfigMap` definition:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ovs-flows-config
+  namespace: # CNO's namespace
+spec:
+  ipfix:
+    sharedTarget: <service ip+port>
```

Comment:
This `sharedTarget` meaning here is a little confusing, maybe adding a diagram showing that it will connect to the goflow collector daemonset pod for each node would help (something like what you made [here](https://docs.google.com/document/d/1kzNfTLXhMolu8VTcH0m9hmvFluN9yKMy7TSNIXJVTrA/edit#heading=h.7yb4abpdeo6w))

I don't think this needs to block the kep from merging though and can be changed in a follow up commit if needed